languagetool-server/src/main/java/org/languagetool/server/TextChecker.java

```diff
@@ -747,34 +778,35 @@ private List<CheckResults> getRuleMatches(AnnotatedText aText, Language lang,
     } else {
       List<CheckResults> res = new ArrayList<>();
 
-      if (preferredLangs.size() < 2 || parameters.get("multilingual") == null || parameters.get("multilingual").equals("false")) {
+      if (/*preferredLangs.size() < 2 ||*/ parameters.get("multilingual") == null || parameters.get("multilingual").equals("false")) {
         res.addAll(getPipelineResults(aText, lang, motherTongue, params, userConfig, listener));
-      } else {
-        // support for multilingual texts:
-        try {
-          Language mainLang = getLanguageVariantForCode(detLang.getDetectedLanguage().getShortCode(), preferredVariants);
-          List<Language> secondLangs = new ArrayList<>();
-          for (String preferredLangCode : preferredLangs) {
-            if (!preferredLangCode.equals(mainLang.getShortCode())) {
-              secondLangs.add(getLanguageVariantForCode(preferredLangCode, preferredVariants));
-              break;
-            }
-          }
-          LanguageAnnotator annotator = new LanguageAnnotator();
-          List<FragmentWithLanguage> fragments = annotator.detectLanguages(aText.getPlainText(), mainLang, secondLangs);
-          List<Language> langs = new ArrayList<>();
-          langs.add(mainLang);
-          langs.addAll(secondLangs);
-          Map<Language, AnnotatedTextBuilder> lang2builder = getBuilderMap(fragments, new HashSet<>(langs));
-          for (Map.Entry<Language, AnnotatedTextBuilder> entry : lang2builder.entrySet()) {
-            res.addAll(getPipelineResults(entry.getValue().build(), entry.getKey(), motherTongue, params, userConfig, listener));
-          }
-        } catch (Exception e) {
-          log.error("Problem with multilingual mode (preferredLangs=" + preferredLangs+ ", preferredVariants=" + preferredVariants + "), " +
-            "falling back to single language.", e);
-          res.addAll(getPipelineResults(aText, lang, motherTongue, params, userConfig, listener));
-        }
-      }
+      } 
+//      else {
```

Comment:
Can we add a note about why this code is commented out, or should it be removed and replaced with an error when the multilingual parameter is set?